probe-rs/src/core/registers.rs

```diff
@@ -13,25 +17,86 @@ pub enum RegisterDataType {
     FloatingPoint,
 }
 
-/// Describes a register with its properties.
+/// This is used to label the register with a specific role that it plays during program execution and exception handling.
+/// This denotes the purpose of a register (e.g. `return address`),
+/// while the [`CoreRegister::name`] will contain the architecture specific label of the register.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub(crate) enum RegisterRole {
+    /// Argument/Result registers like "A0", "a1", "r2", etc. (uses architecture specific names)
+    Argument(&'static str),
+    ProgramCounter,
+    FramePointer,
+    StackPointer,
+    MainStackPointer,
+    ProcessStackPointer,
+    ProcessorStatus,
+    ReturnAddress,
+    FloatingPoint,
+    FloatingPointStatus,
+    Other(&'static str),
+}
+
+impl Display for RegisterRole {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RegisterRole::Argument(name) => write!(f, "{}", name),
+            RegisterRole::ProgramCounter => write!(f, "PC"),
+            RegisterRole::FramePointer => write!(f, "FP"),
+            RegisterRole::StackPointer => write!(f, "SP"),
+            RegisterRole::MainStackPointer => write!(f, "MSP"),
+            RegisterRole::ProcessStackPointer => write!(f, "PSP"),
+            RegisterRole::ProcessorStatus => write!(f, "PSR"),
+            RegisterRole::ReturnAddress => write!(f, "LR"),
+            RegisterRole::FloatingPoint => write!(f, "FPU"),
+            RegisterRole::FloatingPointStatus => write!(f, "FPSR"),
+            RegisterRole::Other(name) => write!(f, "{}", name),
+        }
+    }
+}
+
+/// Describes a core (or CPU / hardware) register with its properties.
+/// Each architecture will have a set of general purpose registers, and potentially some special purpose registers. It also happens that some general purpose registers can be used for special purposes. For instance, some ARM variants allows the `LR` (link register / return address) to be used as general purpose register `R14`."
 #[derive(Debug, Clone, PartialEq)]
-pub struct RegisterDescription {
+pub struct CoreRegister {
+    /// The architecture specific name of the register. This may be identical, or similar to [`RegisterRole`].
     pub(crate) name: &'static str,
-    pub(crate) _kind: RegisterKind,
     pub(crate) id: RegisterId,
-    pub(crate) _type: RegisterDataType,
+    /// If the register plays a special role during program execution and exception handling, this field will be set to the role.
+    /// Otherwise, it will be `None` and the register is a general purpose register.
+    pub(crate) role: Option<RegisterRole>,
+    pub(crate) data_type: RegisterDataType,
```

Comment:
Would it make sense to combine this with `size_in_bits`? They seem very closely related.

Reply:
@Tiwalun Are you thinking something like ???...
```
pub enum RegisterDataType {
    /// Unsigned integer data
    UnsignedInteger(usize),
    /// Floating point data
    FloatingPoint(usize),
}
```

Reply:
This sounds good to me :)

Reply:
Done :)